lib/components/schematic-table.ts

```diff
@@ -0,0 +1,47 @@
+import { distance } from "circuit-json"
+import { z } from "zod"
+import { ninePointAnchor } from "lib/common/ninePointAnchor"
+import { expectTypesMatch } from "lib/typecheck"
+
+export const schematicTableCellProps = z.object({
+  text: z.string(),
+  horizontalAlign: z.enum(["left", "center", "right"]).optional(),
+  verticalAlign: z.enum(["top", "middle", "bottom"]).optional(),
+  fontSize: distance.optional(),
+})
+
+export interface SchematicTableCellProps {
+  text: string
+  horizontalAlign?: "left" | "center" | "right"
+  verticalAlign?: "top" | "middle" | "bottom"
+  fontSize?: number | string
+}
+
+expectTypesMatch<
+  SchematicTableCellProps,
+  z.input<typeof schematicTableCellProps>
+>(true)
+
+export const schematicTableProps = z.object({
+  schX: distance.optional(),
+  schY: distance.optional(),
+  rows: z.array(z.array(schematicTableCellProps)),
+  columnWidths: z.array(distance),
+  rowHeights: z.array(distance),
```

Comment:
This doesn't feel ergonomic, for circuit-json it's ok because it could be computed.

You can add `fontSize` on a per-cell basis as well as the table's overall `fontSize` (both optional)

```suggestion
```

Reply:
```suggestion
  columnWidths: z.array(distance).optional(),
  rowHeights: z.array(distance).optional()
```

Reply:
really how most people would design this is probably:

`<schematictable>
<schematicrow>
<schematiccell>

</schematiccell>
</schematicrow>
</schematictable>
`